Lib/test/test_pkgutil.py

```diff
@@ -549,22 +550,26 @@ def test_loader_deprecated(self):
         with self.check_deprecated():
             pkgutil.ImpLoader("", "", "", "")
 
-    def test_get_loader_avoids_emulation(self):
-        with check_warnings() as w:
-            self.assertIsNotNone(pkgutil.get_loader("sys"))
-            self.assertIsNotNone(pkgutil.get_loader("os"))
-            self.assertIsNotNone(pkgutil.get_loader("test.support"))
-            self.assertEqual(len(w.warnings), 0)
+    def test_get_loader_is_deprecated(self):
+        for module in ["sys", "os", "test.support"]:
+            with check_warnings((
+                "`pkgutil.get_loader` is deprecated since Python 3.12; "
+                "this function is slated for removal in Python 3.14, "
+                "use `importlib.util.find_spec` instead",
+                DeprecationWarning,
+            )):
+                res = pkgutil.get_loader(module)
+            self.assertIsNotNone(res)
 
     @unittest.skipIf(__name__ == '__main__', 'not compatible with __main__')
     def test_get_loader_handles_missing_loader_attribute(self):
         global __loader__
         this_loader = __loader__
         del __loader__
         try:
-            with check_warnings() as w:
+            with warnings.catch_warnings():
+                warnings.simplefilter('ignore', DeprecationWarning)
```

Comment:
Instead of this, you could just decorate the test with `ignore_warnings` from `warnings_helper`, if you wanted.

```
@ignore_warnings(category=DeprecationWarning)